src/main/java/com/google/api/codegen/config/GrpcGapicRetryMapping.java

```diff
@@ -0,0 +1,144 @@
+      RetryParamsDefinitionProto.Builder rpb = RetryParamsDefinitionProto.newBuilder();
+      rpb.setMaxRetryDelayMillis(Durations.toMillis(rp.getMaxBackoff()));
+      rpb.setInitialRetryDelayMillis(Durations.toMillis(rp.getInitialBackoff()));
+      rpb.setRetryDelayMultiplier(floatToDouble(rp.getBackoffMultiplier()));
+      rpb.setTotalTimeoutMillis(timeout);
+      rpb.setName(pName + "_params");
+      paramsDefMap.putIfAbsent(rpb.getName(), rpb.build());
+
+      // construct retry codes from RetryPolicy
+      RetryCodesDefinitionProto.Builder rcb = RetryCodesDefinitionProto.newBuilder();
+      rcb.setName(pName + "_codes");
+      rp.getRetryableStatusCodesList()
+          .forEach(
+              code -> {
+                rcb.addRetryCodes(code.name());
+              });
+      codesDefMap.putIfAbsent(rcb.getName(), rcb.build());
+
+      // apply specific method config (overwrites) or apply service config to all methods (does
+      // not overwrite method-specific policies)
+      for (Name name : mc.getNameList()) {
+        String serviceBase = name.getService();
+
+        if (!Strings.isNullOrEmpty(name.getMethod())) {
+          String fqn = fullyQualifiedName(serviceBase, name.getMethod());
+          methodCodesMap.put(fqn, rcb.getName());
+          methodParamsMap.put(fqn, rpb.getName());
+          continue;
+        }
+
+        Interface interProto = protoInterfaces.get(serviceBase);
+        interProto
+            .getMethods()
+            .forEach(
+                method -> {
+                  String fqn = method.getFullName();
+                  methodCodesMap.putIfAbsent(fqn, rcb.getName());
+                  methodParamsMap.putIfAbsent(fqn, rpb.getName());
+                });
+      }
+    }
+
+    // make immutable
+    ImmutableMap<String, String> codes = ImmutableMap.copyOf(methodCodesMap);
+    ImmutableMap<String, String> params = ImmutableMap.copyOf(methodParamsMap);
+    ImmutableMap<String, RetryCodesDefinitionProto> codesProto = ImmutableMap.copyOf(codesDefMap);
+    ImmutableMap<String, RetryParamsDefinitionProto> paramsProto =
+        ImmutableMap.copyOf(paramsDefMap);
+
+    return new AutoValue_GrpcGapicRetryMapping.Builder()
+        .setMethodCodesMap(codes)
+        .setMethodParamsMap(params)
+        .setCodesDefMap(codesProto)
+        .setParamsDefMap(paramsProto)
+        .build();
+  }
+
+  public abstract ImmutableMap<String, String> methodCodesMap();
+
+  public abstract ImmutableMap<String, String> methodParamsMap();
+
+  public abstract ImmutableMap<String, RetryCodesDefinitionProto> codesDefMap();
+
+  public abstract ImmutableMap<String, RetryParamsDefinitionProto> paramsDefMap();
+
+  public static Builder builder() {
+    return new AutoValue_GrpcGapicRetryMapping.Builder();
+  }
+
+  /** converts a float to a double rounding to the nearest hundredth */
+  private static double floatToDouble(float f) {
+    return Math.round(f * 100) / 100.0;
```

Comment:
Converting `float` to `double` gets extra precision. Why is rounding needed?

Reply:
I realize this is weird, but without rounding it, the value ends up being printed as `1.30000000` or something of the like. I realize its not that big of a deal, but it looked weird. It's just an "aesthetic" issue. Happy to take the rounding out if you think its unnecessary :)

Reply:
Hmmmmm that's very interesting. In that case can you add a comment here to clarify the reason that rounding is needed, just for future reference?

Reply:
I removed it and let Java just convert from float to double. A value of `1.3` in the config is printed as `1.2999999523162842` in the GAPIC. 

This [SO answer](https://stackoverflow.com/questions/916081/convert-float-to-double-without-losing-precision) has some explanation.

I'll add a comment! Thanks

Reply:
Done.